core/src/main/scala/cats/MonadReader.scala

```diff
@@ -0,0 +1,14 @@
+package cats
+
+/** A monad that has the ability to read from an environment. */
+trait MonadReader[F[_, _], R] extends Monad[F[R, ?]] {
+  /** Get the environment */
+  def ask: F[R, R]
+
+  /** Modify the environment */
+  def local[A](f: R => R)(fa: F[R, A]): F[R, A]
```

Comment:
If I have to choose a side in the arg list ordering, I'd probably prefer Chang's.

Partially applying a function to get back a Kleisli => Kleisli seems more natural to me than partially applying a Kleisli and getting back a (R => R) => Kleisli